opencsp/common/lib/deflectometry/ImageProjection.py

```diff
@@ -59,20 +60,83 @@ def __init__(self, size_x: int, size_y: int) -> 'CalParams':
         self.y_screen: np.ndarray[float] = y_mat_screen.flatten()
         # Point index
         self.index: np.ndarray[int] = np.arange(self.x_pixel.size, dtype=int)
+        # Width of Aruco markers
+        self.marker_width: int = int(min(size_x, size_y) * 0.2)
+
+
+@dataclass
+class ImageProjectionData(hdf5_tools.HDF5_IO_Abstract):
+    """Dataclass containting ImageProjection parameters. All position/size units are screen pixels"""
+
+    name: str

```

Comment:
Would you mind including descriptions of each of these parameters? For example:

```python
name: str
""" Experiment name associated with these image projection configuration parameters. """
main_window_size_x: int
"""
Width of the display GUI (pixels). This will typically be the same as
active_area_size_x, but could be different in the case that the window isn't
full screen.
"""
```

As you can probably tell, this will be really helpful for when someone such as myself tries to understand the difference between "main_window_size" and "active_area_size".

Reply:
Yes, this definitely needed some doc strings! Thanks for pointing this out. I added some docs, let me know what you think.

Reply:
These docs are great. Thank you!